test/CMakeLists.txt

```diff
@@ -62,18 +68,29 @@ endif()
 
 if(SINGULARITY_TEST_SESAME)
   target_compile_definitions(eos_tabulated_unit_tests PRIVATE SINGULARITY_TEST_SESAME)
+  target_compile_definitions(closure_unit_tests PRIVATE SINGULARITY_TEST_SESAME)
 endif()
 if(SINGULARITY_TEST_STELLAR_COLLAPSE)
   target_compile_definitions(eos_tabulated_unit_tests
                              PRIVATE SINGULARITY_TEST_STELLAR_COLLAPSE)
 endif()
+if(SINGULARITY_USE_SPINER_WITH_HDF5)
+  target_compile_definitions(closure_unit_tests
+                             PRIVATE SINGULARITY_USE_SPINER_WITH_HDF5)
+endif()
+if(SINGULARITY_BUILD_CLOSURE)
+  target_compile_definitions(closure_unit_tests
+                             PRIVATE SINGULARITY_BUILD_CLOSURE)
```

Comment:
this should already be defined https://github.com/lanl/singularity-eos/blob/main/CMakeLists.txt#L266

Reply:
I'm confused here. Isn't this statement also true for all of the code above this that isn't part of this MR? It seems by this logic we don't need any of the `target_compile_definitions` here at all. If that's the case, I'd prefer this be a separate issue since it goes beyond the code I'm adding here.

Reply:
I'm not sure that they are defined elsewhere so I think some of those definitions are necessary additions. Even if they were duplicated (which from what I can tell they _aren't_), I still think this code should be done the right way and I don't see how the correctness or incorrectness of the code above would result in these requested changes to _your_ change-set outside of the scope of the MR.

Reply:
Ah I understand now and have adopted the following convention:
1. any `SINGULARITY_TEST_*` variables are defined in the CMakeLists.txt file for the tests.
2. any other `SINGULARITY_USE_*` or `SINGULARITY_BUILD_*` variables should be defined in the main project CMakeLists.txt and the only one missing was `SINGULARITY_USE_SPINER_WITH_HDF5`. I've added a compile definition there and I think everything works now.